### ISSUE TITLE
Node::getChildren("<nodename>") key issue.

### ISSUE THREAD

**Issue by Johannestegner** [jitesoft/php-xml2array#1]
On calling `$node->getChildren()` method with name a list is returned.  
The list has the same keys as it would have when it was not filtered, i.e, if first index is removed, the first object in the array will be on index 1 instead of 0 when fetched.  
This is due to using `array_filter` method.  
  
Not really sure if I feel that this is a issue but rather how it should be, but considering adding a optional `perserveKeys = false` argument in the method.

**Comment by DavidBruchmann** [comment]
Not sure if I understand you right, can you give an example?

**Comment by Johannestegner** [comment]
When all children from the node is returned it will be returned as a indexed list, i.e.,:

```
array( 
  [0]=>Node,
  [1]=>Node,
  [2]=>Node
)
```

But when passing a node name as argument, to fetch only children with a given "node name" the list will be returned as an assoc array with number keys, so it will look like an indexed array but will not actually be, as example below.

```
array(
  [1] => Node,
  [3] => Node
)
```

This is due to the `array_filter` method in php, and I personally can not decide if it is a good idea to return the array as this or if it should be reindexed as a indexed array.  
The `preserveKeys` arg would make it possible to choose whether it should be indexed or assoc.

**Comment by DavidBruchmann** [comment]
I understand, I'd see it as feature and keep it like that.
This makes it possible to treat the index theoretical like IDs.

**Comment by Johannestegner** [comment]
Closed this issue and created a new task (#8).
